app/controllers/service_requests_controller.rb

```diff
@@ -188,29 +188,15 @@ def save_and_exit
   end
 
   def add_service
-    existing_service_ids = @service_request.line_items.reject{ |line_item| FINISHED_STATUSES.include?(line_item.status) }.map(&:service_id)
-
-    if existing_service_ids.include?( params[:service_id].to_i )
+    add_service = AddService.new(@service_request,
```

Comment:
While I think it's good to keep the controller method a bit shorter by extracting to a lib file, I think this needs more work to actually improve upon the existing code. This feels like a band-aid more than a fix.

The biggest problem I've had with this controller action is that it's _really slow_ and there's a lot of extra work being done and poor optimization that needs to be improved.